lstchain/image/modifier.py

```diff
@@ -54,32 +54,51 @@ def add_noise_in_pixels(rng, image, extra_noise_in_dim_pixels,
     return image
 
 
-def smear_light_in_pixels(image, camera_geom, smeared_light_fraction):
+def smear_light_in_pixels(rng, image, camera_geom, smeared_light_fraction):
     """
 
     Parameters
     ----------
+    rng: numpy.random.default_rng  random number generator
+
     image: charges (p.e.) in the camera
 
     camera_geom: camera geometry
 
     smeared_light_fraction: fraction of the light in a pixel that will be
-    distributed equally among its immediate surroundings, i.e. immediate
-    neighboring pixels. Some light is lost for pixels which are at the
-    camera edge and hence don't have all possible neighbors
+    distributed among its immediate surroundings, i.e. immediate
+    neighboring pixels, according to Poisson statistics. Some light is
+    lost for pixels which are at the camera edge and hence don't have all
+    possible neighbors
 
     Returns
     -------
     Modified (smeared) image
 
     """
 
-    # Move a fraction of the light in each pixel (fraction) into its neighbors,
-    # to simulate a worse PSF:
-    q_smeared = (image * camera_geom.neighbor_matrix *
-                 smeared_light_fraction / N_PIXEL_NEIGHBORS)
+    # How many p.e. to smear?  Poisson of the smeared light fraction (clipped to a minimum of 0):
+    pe_to_smear = rng.poisson(np.clip(image * smeared_light_fraction, 0, np.inf))
+
+    # How to distribute the smeared charge among neighboring pixels (multinomial):
+    smeared_charges = np.zeros(shape=(len(image), N_PIXEL_NEIGHBORS))
+    for q in np.unique(pe_to_smear):
```

Comment:
This will introduce the same pattern in all pixels that have the same charge "pe_to_smear", correct? Isn't that dangerous?

Reply:
No, not the same pattern. Each one has its own Poisson parameter, and the generator produces according to it.

Reply:
> No, not the same pattern. Each one has its own Poisson parameter, and the generator produces according to it.

Unless I got it wrong (which might well be!)

Reply:
You draw a poisson for each pixel, then only loop over the unique number of photons. That means that all pixels sharing the same `pe_to_smear` will have the exact same distribution of those photons into their neighbors, right?

I think you have to draw multinomial for each pixel like this:


```
neighbor_charges = np.multinomial(pe_to_smear, np.full(6, 1/6))
```

Reply:
So, if e.g. two pixels have 10 p.e. to smear, they will smear the exact same amount of light into the same respective neighbors.

Reply:
> So, if e.g. two pixels have 10 p.e. to smear, they will smear the exact same amount of light into the same respective neighbors.

Just for the record: no, I don't think so, because I use vectorization, one set of "smeared pe's"  is generated for each of the pixels having a given charge q.   Look at the size=num_pixels argument of the call to the multinomial.

Reply:
Ha, right, missed that because of the strange linebreak ;)